src/sql/src/plan/statement/ddl.rs

```diff
@@ -3392,7 +3392,15 @@ pub fn plan_drop_item(
             scx.catalog.resolve_full_name(catalog_entry.name()),
         );
     }
-    if object_type != catalog_entry.item_type() {
+    let item_type = catalog_entry.item_type();
+
+    // Return a more helpful error on `DROP VIEW <recorded-view>`.
+    if object_type == ObjectType::View && item_type == CatalogItemType::RecordedView {
+        bail!(
+            "{} is not a view. Use DROP RECORDED VIEW to remove a recorded view.",
+            scx.catalog.resolve_full_name(catalog_entry.name()),
+        );
```

Comment:
This makes the errors look like this:
```
ERROR:  materialize.public.rv is not a view. Use DROP RECORDED VIEW to remove a recorded view.
```

Ideally, we'd want:
```
ERROR:  materialize.public.rv is not a view
HINT:     Use DROP RECORDED VIEW to remove a recorded view.
```

To get a hint included in the pgwire response, adapter calls `AdapterError::hint`. Unfortunately, the `sql` crate only returns `anyhow::Error` (which gets converted to `AdapterError::Unstructured` by the adapter), which doesn't support structured data AFAIK. Short of refactoring the `sql` crate to use a custom error type, I can't think of a non-hacky way to make this work.

Reply:
I personally don't think it's worth making that change in this PR, but it would probably be good to open a separate issue about it.

Reply:
This looks as expected now thanks to Nikhil's `PlanError` changes.